eZ/Publish/Core/IO/IOBinarydataHandler/SiteAccessDependentBinaryDataHandler.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+/**
+ * This file is part of the eZ Publish Kernel package.
+ *
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+namespace eZ\Publish\Core\IO\IOBinarydataHandler;
+
+use eZ\Bundle\EzPublishIOBundle\ApiLoader\HandlerRegistry;
+use eZ\Publish\Core\IO\IOBinarydataHandler;
+use eZ\Publish\Core\MVC\ConfigResolverInterface;
+use eZ\Publish\SPI\IO\BinaryFileCreateStruct;
+
+class SiteAccessDependentBinaryDataHandler implements IOBinaryDataHandler
```

Comment:
Q: Shouldn't this be internal? Consumption should be done via `IOBinaryDataHandler` anyway, right?